### PR TITLE
[WIP] migrate to terraform 0.12.0-beta1

### PR DESCRIPTION
**Pull request by displague**
This is an early Terraform 0.12.0 PR (based on beta1) to verify if the acceptance tests are passing.

README.md changes may be necessary to reflect Terraform 0.12 versioning

Documentation changes may be necessary (in a future PR) to deal with Terraform 0.12 syntax changes for examples.

### CODE REVIEW

**Comment by displague** [comment]
The following failed

* TestAccDataSourceLinodeDomain 
   `Error: errors during refresh: Domain tf-test-3958038677266808963.example was not found`
* TestAccDataSourceLinodeSSHKey 
   `Error: errors during refresh: Linode SSH Key with label tf_test-186788622284669503 was not found`
* TestAccLinodeInstance_configPairUpdate  
   `Error: Error converting config device <nil>`
* TestAccLinodeInstance_configUpdate 

   ```
   panic: interface conversion: interface {} is nil, not map[string]interface {}
   github.com/terraform-providers/terraform-provider-linode/linode.updateInstanceConfigs(0xc0012ec900, 0xc000046b90, 0x41, 0xc001077500, 0x0, 0xbb8, 0xc000aef440, 0xc000aef580, 0xc000aef680, 0xc000aef900, . . .)
   /opt/teamcity-agent/work/c967492f0152f326/src/github.com/terraform-providers/terraform-provider-linode/linode/linode_instance_helpers.go:301 +0x1b57
   github.com/terraform-providers/terraform-provider-linode/linode.resourceLinodeInstanceUpdate(0xc0001585b0, 0x11cb8a0, 0xc0000d78c0, 0x0, 0xc001617b98)
   /opt/teamcity-agent/work/c967492f0152f326/src/github.com/terraform-providers/terraform-provider-linode/linode/resource_linode_instance.go:1197 +0x9d7
   github.com/terraform-providers/terraform-provider-linode/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc0000ff780, 0xc000938230, 0xc0007dd960, 0x11cb8a0, 0xc0000d78c0, 0x104b501, 0xc0012dd2f8, 0xc0016612f0)
   ```

* TestAccLinodeInstance_diskAndConfig 
   `testing.go:568: Step 0 error: config is invalid: Unsupported argument: An argument named "devices" is not expected here. Did you mean to define a block of type "devices"?`

* TestAccLinodeInstance_diskImage 
   ```
    testing.go:568: Step 1 error: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) {
        }
        
        
        (map[string]string) (len=2) {
         (string) (len=24) "disk.0.authorized_keys.#": (string) (len=1) "1",
         (string) (len=24) "disk.0.authorized_keys.0": (string) ""
        }
   ```
* TestAccLinodeInstance_diskPair 
   ```
    testing.go:568: Step 1 error: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) {
        }
        
        
        (map[string]string) (len=2) {
         (string) (len=24) "disk.0.authorized_keys.#": (string) (len=1) "1",
         (string) (len=24) "disk.0.authorized_keys.0": (string) ""
        }
   ```
* TestAccLinodeInstance_disksAndConfigs 
   ```
    testing.go:568: Step 0 error: config is invalid: 2 problems:
        
        - Unsupported argument: An argument named "devices" is not expected here. Did you mean to define a block of type "devices"?
        - Unsupported argument: An argument named "devices" is not expected here. Did you mean to define a block of type "devices"?
   ```
* TestAccLinodeInstance_tag 
   ```
   panic: interface conversion: interface {} is nil, not map[string]interface {}

   goroutine 466 [running]:
   github.com/terraform-providers/terraform-provider-linode/linode.updateInstanceConfigs(0xc00071af00, 0xc000047180, 0x41, 0xc00048b3e0, 0x0, 0xbb8, 0xc0013ed480, 0xc0013ed5c0, 0xc0013ed6c0, 0xc0013ed7c0, ...)
	/opt/teamcity-agent/work/c967492f0152f326/src/github.com/terraform-providers/terraform-provider-linode/linode/linode_instance_helpers.go:301 +0x1b57
   github.com/terraform-providers/terraform-provider-linode/linode.resourceLinodeInstanceUpdate(0xc0002e3030, 0x11cb8a0, 0xc001236c60, 0x0, 0xc0013be518)
	/opt/teamcity-agent/work/c967492f0152f326/src/github.com/terraform-providers/terraform-provider-linode/linode/resource_linode_instance.go:1197 +0x9d7
   ```
* TestAccLinodeInstance_volumeAndConfig 
   ```
    testing.go:568: Step 0 error: config is invalid: Unsupported argument: An argument named "devices" is not expected here. Did you mean to define a block of type "devices"?
   ```
* TestAccLinodeStackscript_codeChange 
   ```
    testing.go:568: Step 1 error: Check failed: Check 6/11 error: linode_stackscript.foobar: Attribute 'user_defined_fields.#' expected "1", got "0"
   ```
* TestAccLinodeVolume_attached 
   ```
    testing.go:568: Step 1 error: config is invalid: Unsupported argument: An argument named "devices" is not expected here. Did you mean to define a block of type "devices"?
   ```
* TestAccLinodeVolume_detached 
   ```
    testing.go:568: Step 0 error: config is invalid: Unsupported argument: An argument named "devices" is not expected here. Did you mean to define a block of type "devices"?
   ```
* TestAccLinodeVolume_reattachedBetweenInstances 
   ```
    testing.go:568: Step 0 error: config is invalid: Unsupported argument: An argument named "devices" is not expected here. Did you mean to define a block of type "devices"?
   ```
* TestAccLinodeInstance_diskResize 
   ```
    testing.go:568: Step 0 error: config is invalid: Unsupported argument: An argument named "devices" is not expected here. Did you mean to define a block of type "devices"?
   ```
* TestAccLinodeInstance_diskSlotReorder 
   ```
    testing.go:568: Step 0 error: config is invalid: Unsupported argument: An argument named "devices" is not expected here. Did you mean to define a block of type "devices"?
   ```
* TestAccLinodeInstance_resize 
   ```
        Error: Error resizing instance 13461050: [400] Linode busy.
   ```

**Comment by displague** [comment]
Related to https://github.com/terraform-providers/terraform-provider-linode/issues/50

**Comment by asauber** [comment]
cc: myself

File: go.mod [hunk 1]
```diff
@@ -1,34 +1,15 @@
...
-	github.com/hashicorp/go-getter v1.0.2 // indirect
 	github.com/hashicorp/go-hclog v0.0.0-20190109152822-4783caec6f2e // indirect
-	github.com/hashicorp/go-plugin v0.0.0-20190129155509-362c99b11937 // indirect
 	github.com/hashicorp/go-uuid v1.0.1 // indirect
 	github.com/hashicorp/hcl v1.0.0 // indirect
-	github.com/hashicorp/hcl2 v0.0.0-20190130225218-89dbc5eb3d9e // indirect
-	github.com/hashicorp/hil v0.0.0-20190129155652-59d7c1fee952 // indirect
-	github.com/hashicorp/logutils v1.0.0 // indirect
-	github.com/hashicorp/terraform v0.11.12-beta1.0.20190214175014-182daa619826
+	github.com/hashicorp/terraform v0.12.0-beta1
```
Comment (davewongillies) [inline go.mod]: Terraform is now up to 0.12.2, it'd be great update this line to that.

**Comment by Gurpartap** [comment]
@davewongillies Did you try the proposed branch with terraform 0.12.2?

**Comment by displague** [comment]
Closed in favor of #55

**Pull request closed without merging**